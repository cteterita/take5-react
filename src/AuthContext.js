import React from 'react';

const AuthContext = React.createContext({
  currentUser: null,
  signInWithGoogle: null,
  signOut: null,
});

export default AuthContext;
