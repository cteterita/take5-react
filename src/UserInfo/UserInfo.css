.user-info-holder {
  display: flex;
  flex-direction: column;
  min-width: 200px;
}

.user-info-holder button {
  width: 200px;
  margin: auto;
}
