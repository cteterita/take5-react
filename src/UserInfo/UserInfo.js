import React, { useContext } from 'react';
import AuthContext from '../AuthContext';
import './UserInfo.css';

function UserInfo() {
  const auth = useContext(AuthContext);
  if (!auth.currentUser) {
    return null;
  }
  return (
    <div className="user-info-holder">
      <span>
        Logged in as:
        <br />
        {auth.currentUser.email || 'Anonymous'}
      </span>
      <button type="button" onClick={auth.signOut}>Log out</button>
      <button type="button" onClick={auth.upgradeAnonymousAccount} hidden={!auth.currentUser.isAnonymous}>Sign in with Google</button>
    </div>
  );
}

export default UserInfo;
