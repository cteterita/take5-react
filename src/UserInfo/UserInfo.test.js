import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import UserInfo from './UserInfo';
import AuthContext from '../AuthContext';

const testAuthLoggedOut = {
  currentUser: null,
  signOut: () => {},
  upgradeAnonymousAccount: () => {},
};

const testAuthAnonymous = {
  currentUser: {},
  signOut: () => {},
  upgradeAnonymousAccount: () => {},
};

const testAuthGoogle = {
  currentUser: {
    email: 'example@user.com',
  },
  signOut: () => {},
  upgradeAnonymousAccount: () => {},
};

describe('LandingView component', () => {
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <AuthContext.Provider value={testAuthLoggedOut}>
        <UserInfo />
      </AuthContext.Provider>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders as expected while logged out', () => {
    const tree = renderer.create(
      <AuthContext.Provider value={testAuthLoggedOut}>
        <UserInfo />
      </AuthContext.Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders as expected while logged in anonymously', () => {
    const tree = renderer.create(
      <AuthContext.Provider value={testAuthAnonymous}>
        <UserInfo />
      </AuthContext.Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders as expected while logged in using Google', () => {
    const tree = renderer.create(
      <AuthContext.Provider value={testAuthGoogle}>
        <UserInfo />
      </AuthContext.Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
