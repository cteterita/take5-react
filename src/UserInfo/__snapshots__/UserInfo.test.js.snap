// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingView component renders as expected while logged in anonymously 1`] = `
<div
  className="user-info-holder"
>
  <span>
    Logged in as:
    <br />
    Anonymous
  </span>
  <button
    onClick={[Function]}
    type="button"
  >
    Log out
  </button>
  <button
    hidden={true}
    onClick={[Function]}
    type="button"
  >
    Sign in with Google
  </button>
</div>
`;

exports[`LandingView component renders as expected while logged in using Google 1`] = `
<div
  className="user-info-holder"
>
  <span>
    Logged in as:
    <br />
    example@user.com
  </span>
  <button
    onClick={[Function]}
    type="button"
  >
    Log out
  </button>
  <button
    hidden={true}
    onClick={[Function]}
    type="button"
  >
    Sign in with Google
  </button>
</div>
`;

exports[`LandingView component renders as expected while logged out 1`] = `
<div
  className="user-info-holder"
/>
`;
